app/src/main/java/com/davidtomas/taskyapp/features/agenda/presentation/AgendaDetail/AgendaDetailViewModel.kt

```diff
@@ -282,17 +282,33 @@ open class AgendaDetailViewModel(
                     photos = state.photos.plus(
                         PhotoModel(
                             key = UUID.randomUUID().toString(),
-                            imageData = agendaDetailAction.imageByteArray
+                            imageData = agendaDetailAction.imageByteArray,
+                            modificationType = ModificationType.ADD
                         )
                     )
                 )
             }
 
             is AgendaDetailAction.OnPhotoDeleted -> {
-                state = state.copy(
-                    screenMode = ScreenMode.EDIT_ADD,
-                    photos = state.photos.filter { it.key != agendaDetailAction.photoKey }
-                )
+                var deletedPhoto = state.photos.first { it.key == agendaDetailAction.photoKey }
+                if (deletedPhoto.modificationType != ModificationType.ADD) {
```

Comment:
I find `modificationType` as a name a bit misleading if you use it as part of the entity. It's not obvious to someone new seeing this that this is only set for items that couldn't be synced. What about `syncType` or `locallyModifiedType` instead?